src/compiler/checker.ts

```diff
@@ -11634,6 +11634,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             if (symbol.flags & SymbolFlags.ValueModule && !(symbol.flags & SymbolFlags.Assignment)) {
                 return getTypeOfFuncClassEnumModule(symbol);
             }
+
+            if (checkMode !== CheckMode.Normal && checkMode) {
+                return anyType;

```

Comment:
I **think** that there is a non-zero chance that [this PR](https://github.com/microsoft/TypeScript/pull/56429) will get picked up at some point (cc @jakebailey ). If the team decides to go with it then this should return `errorType`:
```suggestion
                return errorType;
```

I think that there is no harm in returning that here already now (unless proven otherwise)

Reply:
Thanks for the comment. Since, it's an error (of course, not a circularity one but still), I think you're right about changing the return value from `anyType` to `errorType`. 

Applied in the last commit.